test/currencyConvertor.ts

```diff
@@ -1,73 +1,175 @@
+import ZeroExExchangeArtifact from '../artifacts/contracts/exchange-wrappers/zeroExExchangeWrapper.sol/zeroExExchangeWrapper.json';
+import { axiosRequest, encode, generateQueryPath } from './helpers';
+import { tokenAbi } from './abis';
 
-const { deployContract } = waffle
+import { CurrencyConvertor } from '../src/types';
+import { MockStarkware } from '../src/types';
+import { ZeroExExchangeWrapper } from '../src/types';
+import _ from 'underscore';
 
 
-chai.use(solidity);
+const { deployContract } = waffle
 const { expect } = chai;
+chai.use(chaiAsPromised)
+
+chai.use(solidity);
+
+const impersonatedAccount: string = '0xf0b2e1362f2381686575265799c5215ef712162f';
+const usdcAddres: string = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48';
+const uniswapAddress: string = '0x1f9840a85d5af5bf1d1762f925bdaddc4201f984';
+
 describe("CurrencyConvertor", () => {
-  let currencyConvertor: any;
-  let zeroExExchangeWrapper: any;
-  let starkware: any;
-  let usdc: any;
+  let currencyConvertor: CurrencyConvertor;
+  let starkware: MockStarkware;
+  let zeroExExchangeWrapper: ZeroExExchangeWrapper;
   beforeEach(async () => {
-    const signers = await ethers.getSigners();
+    // setup account
+    await hre.network.provider.request({
+      method: "hardhat_impersonateAccount",
+      params: [impersonatedAccount],
+    });
+    await hre.network.provider.request({
+      method: "hardhat_setBalance",
+      params: [
+        impersonatedAccount,
+        "0x1000000000000000000",
+      ],
+    });
+    const signer = await ethers.getSigner(impersonatedAccount);
 
-    zeroExExchangeWrapper = await deployContract(
-      signers[0],
-      ZeroExExchangeWrapperArtifact,
-    );
+    // deploy contracts
     starkware = await deployContract(
-      signers[0],
+      signer,
       StarkwareArtifact,
-    );
-    usdc = await deployContract(
-      signers[0],
-      UsdcContractArtifact,
-    )
-
-    await usdc.mint(
-      zeroExExchangeWrapper.address,
-      1000e6,
-    )
-    await usdc.mint(
-      starkware.address,
-      1000e6,
-    )
-
+      [uniswapAddress],
+    ) as MockStarkware;
     currencyConvertor = await deployContract(
-      signers[0],
+      signer,
       CurrencyConvertorArtifact,
       [
         starkware.address,
-        usdc.address, // placeholder for USDC
-        156,
+        uniswapAddress,
+        '1',
       ],
+    ) as CurrencyConvertor;
+    zeroExExchangeWrapper = await deployContract(
+      signer,
+      ZeroExExchangeArtifact,
+    ) as ZeroExExchangeWrapper;
+
+    // get ERC20 contracts
+    const usdcContract = new ethers.Contract(
+    usdcAddres,
+    tokenAbi,
+    signer,
     );
 
-    await usdc.mint(
-      signers[0].address,
-      1000e6,
-    )
+    console.log(await usdcContract.symbol());
 
-    await usdc.approve(currencyConvertor.address, 1e10);
+    // approve ERC20 contracts
+    await usdcContract.approve(
```

Comment:
Kind of confusing to have USDC contract approve another contract to pull funds.. Any reason why you went away from minting USDC to an address?

Reply:
because Im forking mainnet and this is the real USDC contract and Im spying on someone with USDC